Actions/DetermineProjectsToBuild/DetermineProjectsToBuild.psm1

```diff
@@ -238,18 +238,10 @@ function Get-ProjectsToBuild {
                 $projectsToBuild = @($projects | Where-Object { ShouldBuildProject -baseFolder $baseFolder -project $_ -modifiedFiles $modifiedFilesFullPaths })
             }
 
-            if($settings.useProjectDependencies) {
-                $buildAlso = @{}
-
-                # Calculate the full projects order
-                $fullProjectsOrder = AnalyzeProjectDependencies -baseFolder $baseFolder -projects $projects -buildAlso ([ref]$buildAlso) -projectDependencies ([ref]$projectDependencies)
-
-                $projectsToBuild = @($projectsToBuild | ForEach-Object { $_; if ($buildAlso.Keys -contains $_) { $buildAlso."$_" } } | Select-Object -Unique)
-            }
-            else {
-                # Use a flatten build order (all projects on the same level)
-                $fullProjectsOrder = @(@{ 'projects' = $projectsToBuild; 'projectsCount' = $projectsToBuild.Count})
-            }
+            # Calculate the full projects order
+            $buildAlso = @{}
+            $fullProjectsOrder = AnalyzeProjectDependencies -baseFolder $baseFolder -projects $projects -buildAlso ([ref]$buildAlso) -projectDependencies ([ref]$projectDependencies)
```

Comment:
NIT - This function (AnalyzeProjectDependencies) uses [ref] parameters and the 2 ref parameters are initialized as @() just to do this.
Maybe it should instead return 3 values (like we do in other places) to increase readability?
The function is only called in one location.